src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -323,16 +345,152 @@ public async Task<ActionResult> Post(string appId, int? instanceOwnerId, [FromBo
             try
             {
                 Instance result = await _instanceRepository.Create(createdInstance);
-                return Ok(result);
+
+                if (result == null)
+                {
+                    throw new ArgumentNullException("result");
+                }
             }
             catch (Exception e)
             {
                 logger.LogError($"Unable to create {appId} instance for {ownerId} due to {e}");
                 return StatusCode(500, $"Unable to create {appId} instance for {ownerId} due to {e}");
             }
+            
+            Stream theStream = null;
+            string contentFileName = null;
+            string contentType = null;
+            long fileSize = 0;
+            
+            if (MultipartRequestHelper.IsMultipartContentType(Request.ContentType))
+            {
+                MediaTypeHeaderValue mediaType = MediaTypeHeaderValue.Parse(Request.ContentType);
+                string boundary = MultipartRequestHelper.GetBoundary(mediaType, _defaultFormOptions.MultipartBoundaryLengthLimit);
+
+                MultipartReader reader = new MultipartReader(boundary, Request.Body);
+
+                // Execute ReadNextSectionAsync() twice, to expose the second part of the Multipart content,
+                // first part has already been handled in ReadInstanceTemplateFromBody(HttpRequest req).
+                MultipartSection section = reader.ReadNextSectionAsync().Result;
+                section = reader.ReadNextSectionAsync().Result;
+
+                bool hasContentDisposition = ContentDispositionHeaderValue.TryParse(section.ContentDisposition, out ContentDispositionHeaderValue contentDisposition);
```

Comment:
Why is this logic outside the while loop?

Reply:
Fixed.